### PR TITLE
Issue#634 clean up policy in topic list

### PR DESCRIPTION
**Pull request by MarselAhmetov**
adding cleanup policy to InternalTopic and some code rearrange

### CODE REVIEW

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/InternalTopic.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
   private final Map<Integer, InternalPartition> partitions;
   private final List<InternalTopicConfig> topicConfigs;
 
+  private final String cleanUpPolicy;
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/InternalTopic.java:17]: should we use enum for it?

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java [hunk 2]
```diff
@@ -205,12 +205,17 @@ private InternalClusterMetrics collectTopicsMetrics(Map<String, InternalTopic> t
...
-    ).collect(Collectors.toMap(
-        InternalTopic::getName,
-        e -> e
-    ));
+    return topics.stream()
+        .map(t -> t.toBuilder().topicConfigs(configs.get(t.getName())).build())
+        .map(t -> t.toBuilder().cleanUpPolicy(t.getTopicConfigs().stream()
+            .filter(config -> config.getName().equals("cleanup.policy"))
+            .findFirst()
+            .orElseGet(() -> InternalTopicConfig.builder().value(null).build())
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java:213]: Are you returning empty topic if no config value? As for me this is incorrect behaviour

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java (hunk 2 above)
Comment (MarselAhmetov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java:213]: Should I throw exception here if cleanup.policy is not found?

Comment (germanosin) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java:213]: I'll prefer to send unknown in this case

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/CleanupPolicy.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package com.provectus.kafka.ui.model;
+
+import com.provectus.kafka.ui.exception.IllegalEntityStateException;
+import java.util.Arrays;
+
+public enum CleanupPolicy {
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/CleanupPolicy.java:6]: You have these in contract, why do we need it?

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/CleanupPolicy.java (hunk 3 above)
Comment (MarselAhmetov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/CleanupPolicy.java:6]: Here I added method for getting CleanupPolicy enum form string

**Review by germanosin — APPROVED** [review]

**Pull request merged**